scripts/globals/combat/magic_hit_rate.lua

```diff
@@ -122,104 +115,204 @@ xi.combat.magicHitRate.calculateActorMagicAccuracy = function(actor, target, spe
         magicAcc = magicAcc + 256
     end
 
-    -----------------------------------
-    -- magicAcc from Job Points.
-    -----------------------------------
+    return magicAcc
+end
+
+-- Magic Accuracy from Merits.
+local function magicAccuracyFromMerits(actor, skillType, spellElement)
+    local magicAcc = 0
+    local actorJob = actor:getMainJob()
+
     switch (actorJob) : caseof
     {
-        [xi.job.WHM] = function()
-            magicAcc = magicAcc + actor:getJobPointLevel(xi.jp.WHM_MAGIC_ACC_BONUS)
-        end,
-
         [xi.job.BLM] = function()
-            magicAcc = magicAcc + actor:getJobPointLevel(xi.jp.BLM_MAGIC_ACC_BONUS)
+            if skillType == xi.skill.ELEMENTAL_MAGIC then
+                magicAcc = actor:getMerit(xi.merit.ELEMENTAL_MAGIC_ACCURACY)
+            end
         end,
 
         [xi.job.RDM] = function()
-            -- RDM Job Point: During saboteur, Enfeebling MACC +2
+            -- Category 1
             if
-                skillType == xi.skill.ENFEEBLING_MAGIC and
-                actor:hasStatusEffect(xi.effect.SABOTEUR)
+                spellElement >= xi.magic.element.FIRE and
+                spellElement <= xi.magic.element.WATER
             then
-                magicAcc = magicAcc + (actor:getJobPointLevel(xi.jp.SABOTEUR_EFFECT)) * 2
+                magicAcc = actor:getMerit(xi.combat.element.rdmMerit[spellElement])
             end
 
-            -- RDM Job Point: Magic Accuracy Bonus, All MACC + 1
-            magicAcc = magicAcc + actor:getJobPointLevel(xi.jp.RDM_MAGIC_ACC_BONUS)
+            -- Category 2
+            magicAcc = magicAcc + actor:getMerit(xi.merit.MAGIC_ACCURACY)
         end,
 
         [xi.job.NIN] = function()
             if skillType == xi.skill.NINJUTSU then
-                magicAcc = magicAcc + actor:getJobPointLevel(xi.jp.NINJITSU_ACC_BONUS)
+                magicAcc = actor:getMerit(xi.merit.NIN_MAGIC_ACCURACY)
             end
         end,
 
-        [xi.job.SCH] = function()
-            if
-                (spellGroup == xi.magic.spellGroup.WHITE and actor:hasStatusEffect(xi.effect.PARSIMONY)) or
-                (spellGroup == xi.magic.spellGroup.BLACK and actor:hasStatusEffect(xi.effect.PENURY))
-            then
-                magicAcc = magicAcc + actor:getJobPointLevel(xi.jp.STRATEGEM_EFFECT_I)
+        [xi.job.BLU] = function()
+            if skillType == xi.skill.BLUE_MAGIC then
+                magicAcc = actor:getMerit(xi.merit.MAGICAL_ACCURACY)
```

Comment:
Not a review comment per se:

`xi.merit.MAGICAL_ACCURACY` ? is this right? Seems to be shared with RDM? Are they the same mod?

Could be, given you can't have a main job of RDM and BLU at the same time. But it appears NIN is different.